specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/VirtualNetworkCreateSubnetWithDelegation.json

```diff
@@ -0,0 +1,113 @@
+{
+  "parameters" : {
+    "api-version": "2018-08-01",
+    "subscriptionId" : "subId",
+    "resourceGroupName" : "rg1",
+    "virtualNetworkName" : "test-vnet",
+    "parameters": {
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "10.0.0.0/16"
+          ]
+        },
+        "subnets": [
+          {
+            "name": "test-1",
+            "properties": {
+              "addressPrefix": "10.0.0.0/24",
+              "delegations": [
+                {
+                  "name": "myDelegation",
+                  "properties": {
+                    "serviceName": "Microsoft.Provider/resourceType"
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+  },
+  "responses" : {
+    "200" : {
+      "body" : {
+        "name" : "test-vnet",
+        "id" : "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/test-vnet",
+        "type" : "Microsoft.Network/virtualNetworks",
+        "location" : "westus",
+        "properties" : {
+          "provisioningState" : "Succeeded",
+          "addressSpace" : {
+            "addressPrefixes" : [
+              "10.0.0.0/16"
+            ]
+          },
+          "subnets": [
+            {
+              "id": "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/test-1",
+              "name": "test-1",
+              "properties": {
+                "addressPrefix": "10.0.0.0/24",
+                "provisioningState": "Succeeded",
+                "delegations": [
+                  {
+                    "name": "myDelegation",
+                    "id": "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/test-1/delegations/myDelegation",
+                    "properties": {
+                      "provisioningState": "Succeeded",
+                      "serviceName": "Microsoft.Provider/resourceType",
+                      "actions": []
+                    }
+                  }
+                ],
+                "purpose": ""

```

Comment:
"purpose" property isn't defined in the spec.

Reply:
Purpose is a readonly property on the subnet that will be set when a delegation is set.